network/shuffle/src/main/java/org/apache/spark/network/shuffle/ExternalShuffleClient.java

```diff
@@ -138,6 +139,26 @@ public void registerWithShuffleServer(
     client.sendRpcSync(registerMessage, 5000 /* timeoutMs */);
   }
 
+  /**
+   * Removes this application from the external shuffle server and optionally deletes local
+   * files.
+   *
+   * @param host Host of the shuffle server.
+   * @param port Port of the shuffle server.
+   * @param cleanupLocalDirs True if corresponding shuffle files should be deleted
+   * @throws IOException
+   */
+  public void applicationRemoved(String host,
+      int port,
```

Comment:
This is really strange. When I have a callback called `applicationRemoved` I would expect to see an appID, not a host and a port. All of that should be transparent to the caller and handled internally right? The other thing is that there's already similar logic in `ExternalShuffleBlockResolver#applicationRemoved`.